conditions.go

```diff
@@ -88,14 +89,46 @@ func (a Address) Equals(b Address) bool {
 // MarshalJSON provides a hex representation for JSON,
 // to override the standard base64 []byte encoding
 func (a Address) MarshalJSON() ([]byte, error) {
-	return marshalHex(a)
+	s := strings.ToUpper(hex.EncodeToString(a))
+	return json.Marshal(s)
 }
 
-// UnmarshalJSON parses JSON in hex representation,
-// to override the standard base64 []byte encoding
-func (a *Address) UnmarshalJSON(src []byte) error {
-	dst := (*[]byte)(a)
-	return unmarshalHex(src, dst)
+func (a *Address) UnmarshalJSON(raw []byte) error {
+	var enc string
+	if err := json.Unmarshal(raw, &enc); err != nil {
+		return errors.Wrap(err, "cannot decode json")
+	}
+
+	// If the encoded string starts with a prefix, cut it off and use
+	// specified decoding method instead of default one.
+	chunks := strings.SplitN(enc, ":", 2)
```

Comment:
This raised a flag for me at first, but I guess it is right.

`:` will never occur in the non-prefixed hex-case, and in every other case, there must be a prefix.
Fair enough